httpx/dispatch/http2.py

```diff
@@ -158,6 +159,17 @@ def initiate_connection(self) -> None:
             if isinstance(event, h2.events.DataReceived):
                 self.h2_state.acknowledge_received_data(len(event.data), stream_id)
```

Comment:
I think that we should to replace `len(event.data)` with `event.flow_controlled_length`, according to [this](https://python-hyper.org/projects/h2/en/stable/advanced-usage.html#working-with-flow-control) part of the h2 documentation, and I also don’t quite understand why the code below with manual control of the window size is required, since the `acknowledge_received_data` should do it by itself as written [here](https://python-hyper.org/projects/h2/en/stable/advanced-usage.html#auto-flow-control).

Reply:
Implemented this in commit.